src/coresyf_image_crop/tests/test_image_crop.py

```diff
@@ -37,3 +36,19 @@ def test_get_shapefile_crs(self):
         epsg_code = get_shapefile_crs(self.data_source)
         self.assertEqual(epsg_code, self.epsg_code)
 
+    def test_apply_buffer_to_polygon(self):
+        buffer_crs = get_raster_crs(gdal.Open(self.test_image_4236))
+        polygon_extent = get_shapefile_polygon_extent(self.data_source)
```

Comment:
The same as before. Additionally, having the input polygon explicitly defined in the test would make easier to understand if the test is correctly written.